conan/internal/runner/docker.py

```diff
@@ -0,0 +1,207 @@
+            os.mkdir(os.path.join(self.abs_runner_home_path, 'profiles'))
+            for file_name in ['global.conf', 'settings.yml', 'remotes.json']:
+                src_file = os.path.join(ConfigAPI(self.conan_api).home(), file_name)
+                if os.path.exists(src_file):
+                    shutil.copy(src_file, os.path.join(self.abs_runner_home_path, file_name))
+            self._copy_profiles(self.args.profile_build)
+            self._copy_profiles(self.args.profile_host)
+            if self.cache == 'copy':
+                tgz_path = os.path.join(self.abs_runner_home_path, 'local_cache_save.tgz')
+                docker_info(f'Save host cache in: {tgz_path}')
+                self.conan_api.cache.save(self.conan_api.list.select(ListPattern("*:*")), tgz_path)
+        return volumes, environment
+
+    def init_container(self):
+        min_conan_version = '2.1'
+        stdout, _ = self.run_command('conan --version', log=True)
+        docker_conan_version = str(stdout.replace('Conan version ', '').replace('\n', '').replace('\r', '')) # Remove all characters and color
+        if Version(docker_conan_version) <= Version(min_conan_version):
+            ConanOutput().status(f'ERROR: conan version inside the container must be greater than {min_conan_version}', fg=Color.BRIGHT_RED)
+            raise ConanException( f'conan version inside the container must be greater than {min_conan_version}')
+        if self.cache == 'shared':
+            self.run_command('mkdir -p ${HOME}/.conan2/profiles', log=False)
+            self.run_command('cp -r "'+self.abs_docker_path+'/.conanrunner/profiles/." ${HOME}/.conan2/profiles/.', log=False)
+            for file_name in ['global.conf', 'settings.yml', 'remotes.json']:
+                if os.path.exists( os.path.join(self.abs_runner_home_path, file_name)):
+                    self.run_command('cp "'+self.abs_docker_path+'/.conanrunner/'+file_name+'" ${HOME}/.conan2/'+file_name, log=False)
+            if self.cache in ['copy']:
+                self.run_command('conan cache restore "'+self.abs_docker_path+'/.conanrunner/local_cache_save.tgz"')
+
+    def update_local_cache(self):
+        if self.cache != 'shared':
+            self.run_command('conan list --graph=create.json --graph-binaries=build --format=json > pkglist.json', log=False)
+            self.run_command('conan cache save --list=pkglist.json --file "'+self.abs_docker_path+'"/.conanrunner/docker_cache_save.tgz')
+            tgz_path = os.path.join(self.abs_runner_home_path, 'docker_cache_save.tgz')
+            docker_info(f'Restore host cache from: {tgz_path}')
+            package_list = self.conan_api.cache.restore(tgz_path)
+
+    def _copy_profiles(self, profiles):
+        cwd = os.getcwd()
+        if profiles:
+            for profile in profiles:
+                profile_path = ProfileLoader.get_profile_path(os.path.join(ConfigAPI(self.conan_api).home(), 'profiles'), profile, cwd)
```

Comment:
I could be wrong here - if we are using the profile _name_, rather than the profile already "composed" by the `create` command - this may result in an incomplete thing being copied if the profile has a `include()` referring to other files - we are only going to copy the referenced file, rather than anything else it might need.

if we propagate the resolved profile from the `conan create` command - we can just do a `profile.dumps()` and save it in the container - that should resolve `includes()`, but I suspect it will also _render_ jinja templates and the likes

a reason this is important is that we may have a profile like this:
```
include(windows-msvc193)
[runner]
type=ssh
host=my-remote-host
```

and then the file `windows-msvc193` is a different profile file that doesn't have any `[runner]` related things. I expect devs to operate like this to avoid repetition

@memsharded  may be able to provide more context here : D

Reply:
This is correct, I initially discussed with @davidsanfal that we might want to transfer the files, because of the same reason, there would be files that would evaluate differently in different machines due to jinja rendering.

But it is totally true that the ``include()`` would be a limitation and fail unexpectedly. We need to think about it.